Dockerfile.centos_client

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Default setting is building on nvidia/cuda:10.1-devel-centos7
+ARG BASE_IMAGE=nvidia/cuda:10.1-devel-centos7
```

Comment:
This Dockerfile seems to be identical with `Dokerfile.client` except for the OS part in image name. Perhaps it can just be extension on `Dockerfile.client`, which takes OS as optional argument?

Reply:
I thought about, as I tried to make the script as similar as possible. But, there are a few differences which, I think, could make a unified Dockerfile a bit messy:
- DEBIAN_FRONTEND environment variable (although having this variable at all seems [discouraged](https://github.com/moby/moby/issues/4032#issuecomment-34597177))
- apt-get / yum difference and the packages that they install
- cmake3 -> cmake rename
- the `.` at the end of the cmake command (this just removes a warning in the centos version)
- the `--ignore-installed` flag in the pip install command, which prevents an error in the centos version caused by numpy already being installed

The apt-get / yum difference seems like the biggest issue here. Do you have a suggestion for how this could be done in a unified Dockerfile?